apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+
+  // Method for updating the header when needed
+  updateComposerHeader: function conv_updateComposerHeader() {
+    var recipientCount = this.recipients.numbers.length;
+    if (recipientCount > 0) {
+      this.setHeaderContent({
+        id: 'recipient',
+        args: { n: recipientCount }
+      });
+    } else {
+      this.setHeaderContent('newMessage');
+    }
+  },
+
+  // scroll position is considered as "manual" if the view is not completely
+  // scrolled to the bottom
+  isScrolledManually: false,
+
+  // We define an edge for showing the following chunk of elements
+  manageScroll: function conv_manageScroll(oEvent) {
+    var scrollTop = this.container.scrollTop;
+    var scrollHeight = this.container.scrollHeight;
+    var clientHeight = this.container.clientHeight;
+
+    this.isScrolledManually = ((scrollTop + clientHeight) < scrollHeight);
+
+    // Check if the banner has been showed and close it when the scroll
+    // reach the bottom
+    if (!this.isScrolledManually && this.isNewMessageNoticeShown) {
+      this.hideNewMessageNotice();
+    }
+
+    // kEdge will be the limit (in pixels) for showing the next chunk
+    var kEdge = 30;
+    if (scrollTop < kEdge) {
+      this.showChunkOfMessages(this.CHUNK_SIZE);
+      // We update the scroll to the previous position
+      // taking into account the previous offset to top
+      // and the current height due to we have added a new
+      // chunk of visible messages
+      this.container.scrollTop =
+        (this.container.scrollHeight - scrollHeight) + scrollTop;
+    }
+  },
+
+  scrollViewToBottom: function conv_scrollViewToBottom() {
+    if (!this.isScrolledManually &&
+        this.container.lastElementChild &&
+        Navigation.isCurrentPanel('thread')) {
+      this.container.lastElementChild.scrollIntoView(false);
+    }
+  },
+
+  forceScrollViewToBottom: function conv_forceScrollViewToBottom() {
+    this.isScrolledManually = false;
+    this.scrollViewToBottom();
+  },
+
+  showNewMessageNotice: function conv_showNewMessageNotice(message) {
+    Contacts.findByPhoneNumber(message.sender).then((contacts) => {
+      var sender = message.sender;
+      if (contacts && contacts.length) {
+        var details = Utils.getContactDetails(sender, contacts[0]);
+        sender = details.title || sender;
+        // Get the first name
+        var index = sender.indexOf(' ');
+        if (index !== -1) {
+          sender = sender.slice(0, index);
+        }
+      }
+
+      var notice = this.newMessageNotice;
+      var newMessageContactNode = notice.querySelector('.contact');
+      newMessageContactNode.textContent = sender;
+
+      this.isNewMessageNoticeShown = true;
+      notice.classList.remove('hide');
+    });
+  },
+
+  hideNewMessageNotice: function conv_hideNewMessageNotice() {
+    this.isNewMessageNoticeShown = false;
+    //Hide the new message's banner
+    this.newMessageNotice.classList.add('hide');
+  },
+
+  close: function conv_close() {
```

Comment:
> isKeyboardDisplayed, _ensureKeyboardIsHidden can probably be somewhere in shared code, utils maybe (although don't really want to make utils.js catch-all module)?

Agreed these are more common utils.

> close, back, _onNavigatingBack - once we remove draft discard/save prompt, the code will become easier and won't look that scary/big to be duplicated.

Wondering if we could have a mixin for draft handling (just throwing the idea, I don't have much idea about how it could look like -- or even if it's needed given we want to remove the prompt).

maybe we should remove the prompt first, especially that Jenny was OK with it before she left.